src/native/libs/System.Security.Cryptography.Native/pal_evp_cipher.c

```diff
@@ -123,36 +123,12 @@ void CryptoNative_EvpCipherDestroy(EVP_CIPHER_CTX* ctx)
     }
 }
 
-int32_t CryptoNative_EvpCipherReset(EVP_CIPHER_CTX* ctx)
-{
-    // EVP_CipherInit_ex with all nulls preserves the algorithm, resets the IV,
-    // and maintains the key.
-    //
-    // The only thing that you can't do is change the encryption direction,
-    // that requires passing the key and IV in again.
-    //
-    // But since we have a different object returned for CreateEncryptor
-    // and CreateDecryptor we don't need to worry about that.
-    uint8_t* iv = NULL;
-
-#ifdef NEED_OPENSSL_3_0
-    // OpenSSL 3.0 alpha 13 does not properly reset the IV. Work around that by
-    // asking for the original IV, and giving it back.
-    uint8_t tmpIV[EVP_MAX_IV_LENGTH];
-
-    // If we're direct against 3.0, or we're portable and found 3.0
-    if (API_EXISTS(EVP_CIPHER_CTX_get_original_iv))
-    {
-        if (EVP_CIPHER_CTX_get_original_iv(ctx, tmpIV, sizeof(tmpIV)) != 1)
-        {
-            return 0;
-        }
-
-        iv = tmpIV;
-    }
-#endif
+int32_t CryptoNative_EvpCipherReset(EVP_CIPHER_CTX* ctx, uint8_t* pIv, int32_t cIv)
+{
+    assert(cIv >= 0 && (pIv != NULL || cIv == 0));
+    (void)cIv;
```

Comment:
We technically don't _have_ to pass the IV length to the shims, either for Android or OpenSSL - but it seemed weird not to. If we want to get rid of the `cIv` parameter, that's fine too.